src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -56,13 +57,22 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner = owner;
                 _subjectBuffer = subjectBuffer;
 
+                // track context changes
+                _registration = Workspace.GetWorkspaceRegistration(_subjectBuffer.AsTextContainer());
+                _registration.WorkspaceChanged += OnWorkspaceChanged;
+
+                if (_registration.Workspace != null)
+                {
+                    // in unit test, it is possible, we create tagger for buffer which doesn't have associated document
+                    _registration.Workspace.DocumentActiveContextChanged += OnDocumentActiveContextChanged;
+                }
+
                 var document = _subjectBuffer.AsTextContainer().GetOpenDocumentInCurrentContext();
                 _currentDocumentId = document?.Id;
 
                 // Kick off a background task to collect the initial set of diagnostics.
-                var cancellationToken = _initialDiagnosticsCancellationSource.Token;
                 var asyncToken = _owner._listener.BeginAsyncOperation(GetType() + ".GetInitialDiagnostics");
-                var task = Task.Run(() => GetInitialDiagnosticsInBackground(document, cancellationToken), cancellationToken);
```

Comment:
not passing this token is not good.

Reply:
it is field. doesn't need to pass in. it can use field to check.